### PR TITLE
chore: extend mongodb compatibility to 6.x

### PR DESCRIPTION
**Pull request by saltire**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Updates the `mongodb` dependency from v5.0.0 to v6.0.0.

I didn't see anything in the [release notes](https://github.com/mongodb/node-mongodb-native/blob/HEAD/etc/notes/CHANGES_6.0.0.md) to warrant changes to the code, so this PR just updates the MongoDB and Node versions, following the PR for v5.

- **Checklist:**

- [n/a] Added test cases
- [x] Updated changelog

### CODE REVIEW

**Comment by saltire** [comment]
@jdesboeufs @mingchuno

**Comment by saltire** [comment]
Added fix for failed build.

**Comment by saltire** [comment]
Looks like the build needs to be manually approved, any maintainer out there willing to run it? @jdesboeufs @mingchuno

**Comment by thuh-reflact** [comment]
> Looks like the build needs to be manually approved, any maintainer out there willing to run it? @jdesboeufs @mingchuno

That would be great.... GOGOGO

**Comment by apydo** [comment]
@MarkoCen, could you please approve it and merge ?

**Comment by jdesboeufs** [comment]
I will look at this.

**Comment by krzysztof-cislo** [comment]
Any chance to merge this PR soon and create new release?

**Comment by jdesboeufs** [comment]
Since `mongodb@6.0` introduces a big breaking change (`findOneAndX` not returning metadata by default), I'm not sure we should "force" having `mongodb >= 6.0`.

This module doesn't use specific `mongodb@6.0` features. What do you think?

File: .github/workflows/sanity.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x]
```
Comment (jdesboeufs) [inline .github/workflows/sanity.yml:15]: We have to keep this for now

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [6.0.0] - 2023-08-28
```
Comment (jdesboeufs) [inline CHANGELOG.md:9]: It's not your responsability to release :)
The next release will be `5.1.0`!

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### **BREAKING CHANGES**
+
+- Upgraded peer dependency `mongodb` to 6.0.0
```
Comment (jdesboeufs) [inline CHANGELOG.md:13]: Not needed. See other comment.

File: README.md [hunk 4]
```diff
@@ -49,7 +49,7 @@ yarn add connect-mongo
 ## Compatibility
 
 * Support Express up to `5.0`
-* Support [native MongoDB driver](http://mongodb.github.io/node-mongodb-native/) `>= 5.0`
+* Support [native MongoDB driver](http://mongodb.github.io/node-mongodb-native/) `>= 6.0`
```
Comment (jdesboeufs) [inline README.md:52]: Not needed

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "connect-mongo",
-  "version": "5.0.0",
+  "version": "6.0.0",
```
Comment (jdesboeufs) [inline package.json:3]: Please let this unchanged

File: package.json [hunk 6]
```diff
@@ -54,11 +54,11 @@
     "prepare-release": "run-s reset-hard test cov:check doc:html version doc:publish"
   },
   "engines": {
-    "node": ">=12.9.0"
+    "node": ">=16.20.0"
```
Comment (jdesboeufs) [inline package.json:57]: Not needed for now. We will drop versions prior 16 LTS in the next major release.

File: package.json [hunk 7]
```diff
@@ -54,11 +54,11 @@
...
   },
   "peerDependencies": {
     "express-session": "^1.17.1",
-    "mongodb": "^5.1.0"
+    "mongodb": "^6.0.0"
```
Comment (jdesboeufs) [inline package.json:61]:
```suggestion
    "mongodb": ">= 5.0.0 < 7"
```

**Review by jdesboeufs — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I suggest some changes so we will be able to release a minor version quickly, and then we will release a major dropping old Node.js versions.

**Comment by saltire** [comment]
@jdesboeufs Thanks for reviewing! Implemented changes, please take a look.

**Review by jdesboeufs — APPROVED** [review]

**Comment by saltire** [comment]
Hmm do we have to drop Node 14 if we want to support mongodb v6?

**Comment by jdesboeufs** [comment]
No but sanity workflo must be updated according to relaxed peer dependency.
I'm working on this!

File: package.json [hunk 8]
```diff
@@ -58,7 +58,7 @@
   },
   "peerDependencies": {
     "express-session": "^1.17.1",
-    "mongodb": "^5.1.0"
+    "mongodb": ">= 5.0.0 < 7"
```
Comment (jdesboeufs) [inline package.json:61]: I made a mistake here
```suggestion
    "mongodb": ">= 5.1.0 < 7"
```

File: package.json [hunk 9]
```diff
@@ -95,7 +95,7 @@
     "gh-pages": "^3.1.0",
     "husky": "4",
     "lint-staged": "^10.5.4",
-    "mongodb": "^5.1.0",
+    "mongodb": "^6.0.0",
```
Comment (jdesboeufs) [inline package.json:98]: Finally please let this part unchanged at the moment.

**Review by jdesboeufs — CHANGES_REQUESTED** [review]
And to test against `mongodb@6` I suggest you to add a new step in `sanity.yml`.
Something like this:
```yaml
- run: yarn add mongodb@^6 && yarn test
  if: ${{ matrix.node-version !== '4.x' }}
```

**Comment by jdesboeufs** [comment]
If you prefer I can finish. No problem with this :)

**Comment by saltire** [comment]
OK, updated. Not sure if the extra CI step is in the right place, let me know if it needs fixing.

File: .github/workflows/sanity.yml [hunk 10]
```diff
@@ -33,6 +33,8 @@ jobs:
             ${{ runner.os }}-yarn-
       - run: yarn install
       - run: yarn test
+      - run: yarn add mongodb@6 && yarn test
+        if: ${{ matrix.node-version }} !== '14.x'
```
Comment (jdesboeufs) [inline .github/workflows/sanity.yml:37]:
```suggestion
        if: ${{ matrix.node-version !== '14.x' }} 
```

**Review by jdesboeufs — CHANGES_REQUESTED** [review]

**Review by jdesboeufs — APPROVED** [review]

**Pull request merged**